src/Components/Web.JS/src/Rendering/BrowserRenderer.ts

```diff
@@ -65,10 +65,10 @@ export class BrowserRenderer {
     if (elementsToClearOnRootComponentRender.delete(element)) {
       emptyLogicalElement(element);
 
-      if (element instanceof Comment) {
+      if (Object.prototype.toString.call(element) === "[object Comment]") {
         // We sanitize start comments by removing all the information from it now that we don't need it anymore
         // as it adds noise to the DOM.
-        element.textContent = '!';
+        (element as unknown as Comment).textContent = '!';
```

Comment:
This check is incorrect, the right check should be `element instanceof element.ownerDocument.defaultView.Comment`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof#instanceof_and_multiple_realms

Reply:
Also, rather than spread this logic all over, we should put it on a helper method `isComment` to ensure the check is the same across all the callsites.

Reply:
Tested using `element instanceof element.ownerDocument.defaultView.Comment`. Result is similar to the original Microsoft implementation - the Comment element is NOT detected as such in all cases.

Here is the code and the console output when the error is present:
```
if (e instanceof Comment) {
  return e.parentNode;
} else if (Object.prototype.toString.call(e) === "[object Comment]") {
  console.warn("---> This *was NOT* detected as a Comment, but *is* one. element instanceof element.ownerDocument.defaultView.Comment is ", e instanceof e.ownerDocument.defaultView.Comment)
}

throw new Error("Not a valid logical element ", e)
```
When the error occurs, this is what is printed in the console - notice the `false`:
` ---> This *was NOT* detected as a Comment, but *is* one. Caller is reaching out for the parent node here. Btw element instanceof element.ownerDocument.defaultView.Comment is  false`

Reply:
More console output while stopped at a breakpoint during console.warn above:
![image](https://github.com/user-attachments/assets/c12a439e-a654-48f9-954a-9f5cd1163410)

Reply:
@GStoynev why can't we do `(element instance of Comment || (window.frameElement && element instanceof  window.frameElement.ownerDocument.defaultView.Comment))`

Can you put up a repro as a public github repository to make sure we are looking at the same thing? I would expect `element.ownerDocument` to be the same as `window.frameElement.ownerDocument` when element is inside the iframe.